common/include/km_types.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+/*
+#if defined(_WIN32) || defined(_WIN64)
+#define snprintf _snprintf
+#define vsnprintf _vsnprintf
+#define strcasecmp _stricmp
+#define strncasecmp _strnicmp
+#endif
+*/
+
+#if defined(__LP64__) || defined(_LP64)
+/* 64-bit, g++ */
+#define KMX_64BIT
+#endif
+
+#if defined(_WIN64) && !defined(USE_64)
+/* 64-bit, Windows */
+#define KMX_64BIT
+#endif
+
+typedef uint32_t            KMX_DWORD;
+typedef int32_t             KMX_BOOL;
+typedef uint8_t             KMX_BYTE;
+typedef uint16_t            KMX_WORD;
+
+#if defined(__cplusplus)
+typedef char16_t   km_kbp_cp;
```

Comment:
hooray

Reply:
I'm not sure if I am happy or sad. `char16_t` is so unloved by the standard libraries!

Reply:
use ICU4C instead!

Anyway, it's a happier type than wchar will ever be…

Reply:
Let's have a discussion on using ICU4C in the Keyman Core. I think we stand to gain considerable benefit. Only concern is the additional weight for the WASM build.